docs/secure-guardrails/overview.md

```diff
@@ -0,0 +1,265 @@
+    <td rowspan="4">
+    <ul><li>Receive human-written and Semgrep Assistant remediation guidance\*; you can customize the confidence level at which the AI leaves a comment.</li>
+    <li>Ignore or apply a [1-click autofix](/writing-rules/autofix), if it is available, to findings individually.</li></ul>
+    </td>
+</tr>
+<tr>
+    <td>[All GitLab plans](/semgrep-appsec-platform/gitlab-mr-comments)</td>
+</tr>
+<tr>
+    <td>[All Bitbucket plans](/category/bitbucket-pr-comments)</td>
+</tr>
+<tr>
+    <td>[Azure Devops Cloud](/semgrep-appsec-platform/azure-pr-comments)</td>
+</tr>
+<tr>
+    <td>CLI through <code>pre-commit</code></td>
+    <td>Most terminal emulator apps</td>
+    <td>
+    <ul><li>Receive human-written remediation guidance.</li>
+    <li>Ignore or apply [a 1-click autofix](/writing-rules/autofix), if it is available, to findings individually.</li></ul>
+    </td>
+</tr>
+</tbody>
+</table>
+
+_\*To receive Assistant guidance, check that your source code manager (SCM) is supported: [list of supported source code managers (SCMs)](/semgrep-assistant/overview#support-and-availability)._
+
+![Semgrep remediation guidance in VS Code](/img/guardrails-ide-quickfix.png#md-width-centered) <br />
+![Semgrep remediation guidance in VS Code > Quick fix menu](/img/guardrails-ide-fix.png#md-width-centered)
+_**Figure**. Remediation message provided in VS Code. The message appears when a user hovers over findings, which are marked with squiggly lines. Developers can click the **Quick Fix** button to either ignore the finding or, if the rule provides an autofix, apply the fix._
+
+
+![A PR comment detecting a hardcoded secret](/img/guardrails-secrets.png#bordered)
+_**Figure**. A PR comment detecting a hardcoded secret._
+
+### Customizability
+
+Every organization has its own secure coding practices. Customizability ensures that the tool can adapt to the unique needs of an organization.
+
+Semgrep provides customizability through:
+
+- Custom rules - You can create custom rules and deploy them as guardrails. Learn more about Semgrep rule structure in [the next section](#remediation-guidance).
+- Assistant Memories - this feature allows you to add and save additional context when Semgrep Assistant provides remediation. For example, you can provide organization-specific public keys, which Semgrep Assistant remembers.
+
+![Assistant Memories form within a finding's details page](/img/guardrails-memories.png#md-width-centered)
+_**Figure**. A form on a finding's details page where you can enter additional instructions or context._
+
+### Remediation guidance
+
+Remediation guidance can come in three forms:
+
+- The rule's `message`
+- AI-generated remediation guidance through Semgrep Assistant
+- The rule's `fix`
+
+Much of the remediation guidance originates from the rule itself, which is also used to generate Semgrep Assistant's advice (if Assistant is enabled). Learning the basic Semgrep rule structure can help you:
```

Comment:
@pabloest Addresses your original note in previous PR:

> My thought was that going down the path of learning the rule structure is to go down the path of making entirely custom rules, free from AI. I see the point that learning the structure is valuable for AI-assisted things, too.

**This is the only change from the last PR's doc.**